src/harness/unittests/tsserverProjectSystem.ts

```diff
@@ -3167,6 +3167,69 @@ namespace ts.projectSystem {
             host.runQueuedTimeoutCallbacks();
             serverEventManager.checkEventCountOfType("configFileDiag", 3);
         });
+
+        it("are generated when the config file doesnot include file opened but has errors", () => {
+            const serverEventManager = new TestServerEventManager();
+            const file = {
+                path: "/a/b/app.ts",
+                content: "let x = 10"
+            };
+            const file2 = {
+                path: "/a/b/test.ts",
+                content: "let x = 10"
+            };
+            const configFile = {
+                path: "/a/b/tsconfig.json",
+                content: `{
+                    "compilerOptions": {
+                        "foo": "bar",
+                        "allowJS": true
+                    },
+                    "files": ["app.ts"]
+                }`
+            };
+
+            const host = createServerHost([file, file2, libFile, configFile]);
+            const session = createSession(host, {
+                canUseEvents: true,
+                eventHandler: serverEventManager.handler
+            });
+            openFilesForSession([file2], session);
+            serverEventManager.checkEventCountOfType("configFileDiag", 1);
+            for (const event of serverEventManager.events) {
+                if (event.eventName === "configFileDiag") {
+                    assert.equal(event.data.configFileName, configFile.path);

```

Comment:
I think this is a bit confusion for the API users. it would make more sense to not send this error event at all, unless this file is part of the project.